LeaderboardBackend/Services/Impl/EmailSender.cs

```diff
@@ -0,0 +1,62 @@
+using MailKit.Net.Smtp;
+using Microsoft.Extensions.Options;
+using MimeKit;
+
+namespace LeaderboardBackend.Services;
+
+public class EmailSender : IEmailSender
+{
+    private readonly EmailSenderConfig _config;
+    private readonly ILogger<EmailSender> _logger;
+    private readonly ISmtpClient _smtpClient;
+    private readonly MailboxAddress _sender;
+
+    public EmailSender(IOptions<EmailSenderConfig> config, ILogger<EmailSender> logger, ISmtpClient smtpClient)
+    {
+        _config = config.Value;
+        _logger = logger;
+        _smtpClient = smtpClient;
+        _sender = new MailboxAddress(name: _config.SenderName, address: _config.SenderAddress);
+    }
+
+    public async Task SendEmailAsync(string recipientAddress, string subject, string htmlMessage)
+    {
+        if (_config.Smtp is null)
+        {
+            _logger.LogWarning("Can't send email (subject: \"{EmailSubject}\"), SMTP configuration is missing",
```

Comment:
Should this maybe be an error instead of a warning? This seems like it would be a serious issue.

Reply:
changed to error